actors/common/src/main/java/org/sunbird/learner/actors/SchedularActor.java

```diff
@@ -1,41 +1,72 @@
 package org.sunbird.learner.actors;
 
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+
 import org.sunbird.actor.core.BaseActor;
 import org.sunbird.actor.router.ActorConfig;
 import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.exception.ProjectCommonException;
 import org.sunbird.common.models.util.ActorOperations;
 import org.sunbird.common.models.util.BulkUploadActorOperation;
 import org.sunbird.common.models.util.JsonKey;
 import org.sunbird.common.models.util.LoggerEnum;
 import org.sunbird.common.models.util.ProjectLogger;
 import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.quartz.scheduler.OnDemandShadowManager;
 import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
 import org.sunbird.helper.ServiceFactory;
 import org.sunbird.learner.util.UserUtility;
 import org.sunbird.learner.util.Util;
 
+import static org.sunbird.common.request.orgvalidator.BaseOrgRequestValidator.ERROR_CODE;
+
 /** @author Amit Kumar */
 @ActorConfig(
-  tasks = {},
+  tasks = {"onDemandStartScheduler"},
   asyncTasks = {"scheduleBulkUpload"}
 )
 public class SchedularActor extends BaseActor {
+  static final String TYPE = "type";
 
   @Override
   public void onReceive(Request actorMessage) throws Throwable {
     if (actorMessage
         .getOperation()
         .equalsIgnoreCase(ActorOperations.SCHEDULE_BULK_UPLOAD.getValue())) {
       schedule(actorMessage);
+    } else if (actorMessage
```

Comment:
since this actor was created to do bulk upload processing, and i feel it's better to create a separate actor for onDemandScheduler job. 
Some time there might be scenario , this actor is busy with processing upload job and our onDemand job will be kept in mailbox to pick and process later.

Reply:
Right advise, created new one.